### ISSUE TITLE
Lag when Entering Military Base

### ISSUE THREAD

**Issue by estebandellasilva** [CleverRaven/Cataclysm-DDA#51675]
### Describe the bug

I am currently cleaning out a military base (huge amount of red Ms with  a minefield all around)

Everytime i start the game and enter the mine field the game lags a little, whenever i am halfway through the minefield the game lags for like 5-10 seconds(have seen up to 30 seconds) where it looks like the game crashed.

This is repeatable whenever i restart the game.

My Assumption iss that there are probably a lot of Z Levels or something but i rarely had such a huge lag when playing the game. (Labs for example)

### Steps To Reproduce

i am note sure if this iss 100%  reproducable or not but for me its like this

Start the game
Enter the minefield of a military base
Midway through the minefield my character suddenly stops 
wait for x seconds 
move on

### Expected behavior

Lag when entering a Military base should be a lot less in the best case there should be none

### Versions and configuration

Win 10
latest unstable ( i try to keep my game always up to date)
Mods : stats through skills/kills, no fungal

**Comment by Zireael07** [comment]
Upload the save, otherwise it's pretty much a guesswork (MREs? labs? ants?)

**Comment by estebandellasilva** [comment]
[Backup.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/7187893/Backup.zip)

Here you go

**Comment by anoobindisguise** [comment]
I can corroborate this issue, it happens whenever the MRE warehouse is loaded into memory. It is a result of the obscene amount of nested containers most likely, performance degrades heavily while in the area. It's very easy to see the difference between working on a car outside of its influence vs the warehouse being active.

**Comment by StStep** [comment]
oh, i was trying to figure it out too on my save. and it happens when I cross a certain section of the base. Sure enough, all the MRE's are stored on that corner of the base.

**Comment by StStep** [comment]
I created a fresh save with the Overrun start (to spawn in a military base), and I couldn't recreate the lag. Since I saw it in an older save and not a fresh one could it be a compounding slowness that will only show up in larger saves? Maybe only in saves made from an older experimental but loaded in a new one? Potentially a MRE migration thing? I will say trying to open the get menu on a stack of MREs lead to a noticeable delay.

**Comment by estebandellasilva** [comment]
Does the Overrun Szenario start you inside the base? Because what i noticed iss that if you load you safe inside the base the first load time is significantly longer

**Comment by StStep** [comment]
Yeah, you actually start in the warehouse in overrun. I did have the game crash once when doing an overrun start when trying to test this. I'm not sure if that was because it couldn't find a good spawn location or something else.

**Comment by Hirmuolio** [comment]
The problem is the way item processing is currently done.

When item is processed it needs to know things about its container (insulation, etc.). But items do not know anything about their containers.

So when an item that needs processing is in a container then the top level container is processed.
This will cause every item in that container to be also processed.

That is usually not a big problems. Containers usually do not have too many items in them.

MRE packs are not usual items.

![mre](https://user-images.githubusercontent.com/22011552/134736803-6602ad22-e4b9-4e9e-83d2-87b2ee6f92bd.png)

Here we have one unopened MRE pack. All the items in the inventory you see there are in that one MRE. That is a lot of items in one container.
The military base contains probably thousands of those. (quick check says 1000 MRE in single warehouse. ALso other contained foods)

As a quick fix removing the unnecessary packaging fluff would probably halve the number of items and double the performance.

Proper fix would be to rewrite item processing to better handle items in containers.

**Comment by sharpneli** [comment]
I ran into this when profiling. Fresh start in military base -> wait. And a vtune capture from it.

![image](https://user-images.githubusercontent.com/2310634/135765476-35e91287-4146-4ffd-82d6-497d1aa9d868.png)

Vast majority of CPU time is being taken by item processing. It's a recursive algorithm with tons of pointer chasing, in addition to vast majority of the items not actually needing a tick per turn.

In addition to speeding up the handling itself it can make sense to look at how rotting is handled, food doesn't necessarily need to tick every single turn.

**Comment by Hirmuolio** [comment]
Food and temperatures are done in same place and are calculated once every 100 turns.

If the item processing tries to recalculate tempreature/rot faster than that it just skips it.
